ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/api/KeyValueParticipantStateWriter.scala

```diff
@@ -32,9 +32,11 @@ class KeyValueParticipantStateWriter(writer: LedgerWriter, metrics: Metrics) ext
         transactionMeta,
         transaction,
       )
-    val metadata = SimpleCommitMetadata(
```

Comment:
This is the only use of `SimpleCommitMetadata` outside tests. I think we can just delete it and cause a compile failure if someone is using it, rather than making it crash at runtime.

Reply:
Removed `SimpleCommitMetadata` and replaced it with mocks wherever needed in tests.